pkg/lib/cockpit.js

```diff
@@ -3499,24 +3501,23 @@ function factory() {
                 if (watch_channel)
                     return;
 
-                const opts = {
-                    payload: "fswatch1",
-                    path,
-                    superuser: base_channel_options.superuser,
-                };
-                watch_channel = cockpit.channel(opts);
-                watch_channel.addEventListener("message", function (event, message_string) {
-                    let message;
-                    try {
-                        message = JSON.parse(message_string);
-                    } catch (e) {
-                        message = null;
-                    }
-                    if (message && message.path == path && message.tag && message.tag != watch_tag) {
-                        if (options && options.read !== undefined && !options.read)
-                            fire_watch_callbacks(null, message.tag);
-                        else
-                            read();
+                watch_channel = new FsInfoClient(path, ["tag"], { superuser: base_channel_options.superuser });
+                watch_channel.on('change', (state) => {
+                    if (state.error || state.info) {
```

Comment:
I think this check is redundant.  By the time you get a change signal, it is guaranteed that (exactly) one of those should definitely be set.